src/Tribe/PUE/Checker.php

```diff
@@ -518,52 +561,157 @@ private function get_api_message( $info ) {
 				$message = wp_kses( $info->api_invalid_message, 'post' );
 			}
 
-			$message = str_replace( '%plugin_name%', '<b>' . $this->get_plugin_name() . '</b>', $message );
+			$message = str_replace( '%plugin_name%', $this->get_plugin_name(), $message );
 			$message = str_replace( '%plugin_slug%', $this->get_slug(), $message );
 			$message = str_replace( '%update_url%', $this->get_pue_update_url(), $message );
 			$message = str_replace( '%version%', $info->version, $message );
 
 			return $message;
 		}
 
+		private function get_api_update_message() {
+			$plugin_info = $this->plugin_info;
+
+			if ( ! isset( $plugin_info->api_invalid_message ) ) {
+				return false;
+			}
+
+			$message = sprintf(
+				esc_html__( 'There is an update for %s. %sRenew your license%s to get access to bug fixes, security updates, and new features.', 'tribe-common' ),
+				$this->get_plugin_name(),
+				'<a href="https://theeventscalendar.com/license-keys/">',
+				'</a>'
+			);
+
+			return $message;
+		}
+
+		/********************** Admin Notices **********************/
+
 		/**
-		 * Displays an error notice if a premium plugin is activated and the license is expired
+		 * Generates the Notice for plugins with an Invalid or missing license
 		 *
+		 * @return bool|string
 		 * @since 4.3
+		 */
+		public static function is_api_expired_warning() {
+			if ( ! current_user_can( 'install_plugins' ) ) {
+				return false;
+			}
+			remove_action( 'tribe-check-licenses', __CLASS__ . '::setup_warnings' );
+			$results = self::$checkers;
+
+			// Message vars
+			$link   = '<a href="http://m.tri.be/195d" target="_blank">' . esc_html__( 'Visit the Events Calendar website to renew your license.', 'tribe-common' ) . '<span class="screen-reader-text">' .  esc_html__( 'opens in a new window', 'tribe-common' ) . '</span></a>';
+
+			// Begin message
+			$html[] = '<div class="api-check">';
+			$html[] = '<img class="tribe-spirit-animal" src="' . esc_url( Tribe__Main::instance()->plugin_url . 'src/resources/images/spirit-animal.png' ) . '">';
+			$html[] = '<p>' . esc_html__( 'There is an update available for', 'tribe-common' );
+			$html[] = '<span class="plugin-list">';
+
+			foreach ( $results as $plugin ) {
```

Comment:
It looks like this method is hooked up every request but we don't actually find out if there are any plugins with expired licenses until this foreach loop runs - and it will render the notice even if there weren't any such plugins.

Shouldn't we test for results at the start of the method and bail out if there are none? Same basically applies to the `is_api_invalid_warning()` and `is_api_upgrade_warning()` methods.